File: src/main/java/com/server/ordering/service/VerificationService.java
package com.server.ordering.service;

import com.server.ordering.Properties;
import com.server.ordering.domain.MemberType;
import com.server.ordering.domain.PhoneNumber;
import com.server.ordering.repository.PhoneNumberRepository;
import com.twilio.Twilio;
import com.twilio.rest.verify.v2.service.Verification;
import com.twilio.rest.verify.v2.service.VerificationCheck;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.NoResultException;
import javax.persistence.PersistenceException;

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class VerificationService {

    private final PhoneNumberRepository phoneNumberRepository;
    private final Properties pt;

    @Transactional
    public void registerPhoneNumber(PhoneNumber phoneNumber) {
        phoneNumberRepository.save(phoneNumber);
    }

    public boolean isPhoneNumberDuplicated(MemberType memberType, String phoneNumber) throws PersistenceException {
        try {
            phoneNumberRepository.findPhoneNumber(memberType, phoneNumber);
            return true;
        } catch (NoResultException e) {
            return false;
        }
    }

    public void sendCode(String phoneNum) {
        Twilio.init(pt.getAccount_sid(), pt.getAuth_token());
        Verification.creator(pt.getPath_service_sid(), "+82" + phoneNum, "sms")
                .setLocale("ko")
                .create();
    }

    public boolean checkCode(String totalNum, String verificationCode) {
        Twilio.init(pt.getAccount_sid(), pt.getAuth_token());
        VerificationCheck verificationCheck = VerificationCheck.creator(
                        pt.getPath_service_sid(),
                        verificationCode)
                .setTo(totalNum).create();

        return verificationCheck.getValid();
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/service/VerificationService.kt
package com.byeoru.ordering_server.service

import com.byeoru.ordering_server.Properties
import com.byeoru.ordering_server.domain.MemberType
import com.byeoru.ordering_server.repository.PhoneNumberRepository
import com.twilio.Twilio
import com.twilio.rest.verify.v2.service.Verification
import com.twilio.rest.verify.v2.service.VerificationCheck
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import javax.persistence.NoResultException
import javax.persistence.PersistenceException

@Service
@Transactional(readOnly = true)
class VerificationService(private val phoneNumberRepository: PhoneNumberRepository,
                          private val pt: Properties) {

    @Throws(PersistenceException::class)
    fun isPhoneNumberDuplicated(memberType: MemberType, phoneNumber: String): Boolean {
        return try {
            phoneNumberRepository.findPhoneNumber(memberType, phoneNumber)
            true
        } catch (e: NoResultException) {
            false
        }
    }

    fun sendCode(phoneNum: String) {
        Twilio.init(pt.account_sid, pt.auth_token)
        Verification.creator(pt.path_service_sid, "+82$phoneNum", "sms")
            .setLocale("ko")
            .create()
    }

    fun checkCode(totalNum: String, verificationCode: String): Boolean {
        Twilio.init(pt.account_sid, pt.auth_token)
        val verificationCheck = VerificationCheck.creator(pt.path_service_sid, verificationCode)
            .setTo(totalNum).create()
        return verificationCheck.valid
    }
}